magpie/models.py

```diff
@@ -100,32 +110,107 @@ class ExternalIdentity(ExternalIdentityMixin, Base):
+
+    @property
+    def __acl__(self):
+        """
+        Administrators have all permissions, user/group-specific permissions added if user is logged in.
+        """
+        user = self.request.user
+        # allow if role MAGPIE_ADMIN_PERMISSION is somehow directly set instead of inferred via members of admin-group
+        acl = [(Allow, get_constant("MAGPIE_ADMIN_PERMISSION"), ALL_PERMISSIONS)]
+        admins = GroupService.by_group_name(get_constant("MAGPIE_ADMIN_GROUP"), db_session=self.request.db)
+        if admins:
+            # need to add explicit admin-group ALL_PERMISSIONS otherwise views with other permissions than the
+            # default MAGPIE_ADMIN_PERMISSION will be refused access (e.g.: views with MAGPIE_LOGGED_PERMISSION)
+            acl += [(Allow, "group:{}".format(admins.id), ALL_PERMISSIONS)]
+        if user:
+            # user-specific permissions (including group memberships)
+            permissions = UserService.permissions(user, self.request.db)
+            user_acl = permission_to_pyramid_acls(permissions)
+            # allow views that require minimally to be logged in (regardless of who is the user)
+            auth_acl = [(Allow, user.id, Authenticated)]
+            acl += user_acl + auth_acl
+        return acl
+
+
+class UserFactory(RootFactory):
+    def __init__(self, request):
+        super(UserFactory, self).__init__(request)
+        self.path_user = None
+
+    def __getitem__(self, user_name):
+        context = UserFactory(self.request)
+        if user_name == get_constant("MAGPIE_LOGGED_USER", self.request):
+            self.path_user = self.request.user
+        else:
+            self.path_user = UserService.by_user_name(user_name, self.request.db)
+        if self.path_user is not None:
+            self.path_user.__parent__ = self
+            self.path_user.__name__ = user_name
+        context.path_user = self.path_user
+        return context
+
+    @property
+    def __acl__(self):
+        """
+        Grant access to request user according to its relationship to context user (targeted by request path variable).
+
+        If it is the same user (either from explicit name or :py:data:`magpie.constants.MAGPIE_LOGGED_USER` reserved
+        keyword), allow :py:data:`magpie.constants.MAGPIE_LOGGED_PERMISSION` for itself to access corresponding views.
+
+        If request user is unauthenticated, :py:data:`magpie.constants.MAGPIE_LOGGED_USER` or itself, also grant
+        :py:data:`magpie.constants.MAGPIE_CONTEXT_PERMISSION` to allow access to contextually-available details.
+        (e.g.: user can view his own information and public ones)
+        """
+        user = self.request.user
+        acl = super(UserFactory, self).__acl__
+        if user and self.path_user and user.id == self.path_user.id:
+            acl += [(Allow, user.id, get_constant("MAGPIE_LOGGED_PERMISSION")),
+                    (Allow, user.id, get_constant("MAGPIE_CONTEXT_PERMISSION"))]
```

Comment:
I look into permission.rst for more info on CONTEXT_PERMISSION, but I did not find it or anywhere else.
So what is the difference between logged info and contextual info?

Reply:
I agree, `MAGPIE_CONTEXT_PERMISSION` is not well detailed. 
It is a small extension of `MAGPIE_LOGGED_PERMISSION`. 

- MAGPIE_LOGGED_PERMISSION is when path-user == logged-user (but user is authenticated)
- MAGPIE_CONTEXT_PERMISSION is when path-user == logged-user OR path-user refers to unauthorized. 

Example is when unauthenticated user request public permissions using `/users/current/...` or when authenticated user requests public permissions with explicit `/users/anonymous/...`.

Reply:
Ok make sense. I tested it and found something, not sure if this is a bug with context_permission, but if I am not authenticated and try to access /users/(validuserid), I got a valid response... Expecting a not authorized or even a not found to not leak an actual user.

Reply:
Wow. That's weird. Specific tests are evaluating that and they pass... 
I will investigate, definitely not supposed to happen.

Reply:
I don't know if you successfully reproduce that behavior. But if not, maybe it's related to the log out since before trying that I was logged and log out before retrying it.

Reply:
I think I identified the issue. Will had more explicit (and/or revise existing) test cases to validate this situation.